Description: Use Debian libGAP paths
 Debian installs libGAP headers to non-standard paths to avoid conflict with
 the main GAP headers. Moreover, many Sage modules use libGAP, so just add
 these paths to all modules instead of doing module-specific patches.
Author: Jerome BENOIT <calculus@rezozer.net>
Author: Ximin Luo <infinity0@debian.org>
Forwarded: not-needed
--- a/sage/src/sage/libs/gap/test/Makefile
+++ b/sage/src/sage/libs/gap/test/Makefile
@@ -1,6 +1,8 @@
+LIBGAP_CFLAGS = $(shell pkg-config libgap-sage --cflags)
+LIBGAP_LDFLAGS = $(shell pkg-config libgap-sage --libs-only-L)
+LIBGAP_LDLIBS = $(shell pkg-config libgap-sage --libs-only-l)
 
-
-GAPDIR="$(SAGE_LOCAL)/gap/latest"
+GAPDIR="/usr/share/gap"
 
 # VALGRIND=valgrind --leak-check=full
 # VALGRIND=valgrind --db-attach=yes
@@ -12,11 +14,11 @@
 	$(VALGRIND) ./main
 
 main: main.o Makefile
-	gcc -L$(SAGE_LOCAL)/lib -o main main.o -lgap -lcsage -lntl -lstdc++ -lpari -lpython2.7 -lm -lgmp
+	gcc $(LIBGAP_LDFLAGS) -L$(SAGE_LOCAL)/lib -o main main.o $(LIBGAP_LDLIBS) -lcsage -lntl -lstdc++ -lpari -lpython2.7 -lm -lgmp
 
 main.o: main.c
 	echo $(GAPDIR)
-	gcc -std=gnu99 -DGAPDIR="\"$(GAPDIR)\"" -I$(SAGE_LOCAL)/include -I$(SAGE_LOCAL)/include/python2.6 -c -g $^
+	gcc -std=gnu99 -DGAPDIR="\"$(GAPDIR)\"" $(LIBGAP_CFLAGS) -I$(SAGE_LOCAL)/include -I$(SAGE_LOCAL)/include/python2.7 -c -g $^
 
 clean:
 	rm main.o main *~
--- a/sage/src/module_list.py
+++ b/sage/src/module_list.py
@@ -46,6 +46,10 @@
 gsl_library_dirs = gsl_pc['library_dirs']
 gsl_include_dirs = gsl_pc['include_dirs']
 
+# LibGAP
+libgap_pc = pkgconfig.parse('libgap-sage')
+libgap_libs = list(libgap_pc['libraries'])
+
 # GD image library
 gd_pc = pkgconfig.parse('gdlib')
 gd_libs = gd_pc['libraries']
@@ -136,8 +140,8 @@
     "brial_groebner",
     "m4rie",
 ] + m4ri_libs + [
-    "zn_poly", "gap",
-] + gd_libs + png_libs + [
+    "zn_poly",
+] + libgap_libs + gd_libs + png_libs + [
     "m", "readline", "Lfunction" ,
 ] + cblas_libs + [
     "cryptominisat",
@@ -637,15 +641,15 @@
 
     Extension('sage.libs.gap.util',
               sources = ["sage/libs/gap/util.pyx"],
-              libraries = ['gmp', 'gap', 'm']),
+              libraries = ['gmp'] + libgap_libs + ['m']),
 
     Extension('sage.libs.gap.element',
               sources = ["sage/libs/gap/element.pyx"],
-              libraries = ['gmp', 'gap', 'm']),
+              libraries = ['gmp'] + libgap_libs + ['m']),
 
     Extension('sage.libs.gap.libgap',
               sources = ["sage/libs/gap/libgap.pyx"],
-              libraries = ['gmp', 'gap', 'm']),
+              libraries = ['gmp'] + libgap_libs + ['m']),
 
     ###################################
     ##
--- a/sage/src/setup.py
+++ b/sage/src/setup.py
@@ -179,6 +179,12 @@
                 "ntl":     [ os.path.join(SAGE_INC, 'NTL', 'config.h') ]
               }
 
+import pkgconfig
+# LibGAP
+libgap_pc = pkgconfig.parse('libgap-sage')
+libgap_library_dirs = list(libgap_pc['library_dirs'])
+libgap_include_dirs = list(libgap_pc['include_dirs'])
+
 # In the loop below, don't append to any list, since many of these
 # lists are actually identical Python objects. For every list, we need
 # to write (at least the first time):
@@ -196,8 +202,8 @@
 
     m.extra_compile_args = m.extra_compile_args + extra_compile_args
     m.extra_link_args = m.extra_link_args + extra_link_args
-    m.library_dirs = m.library_dirs + [os.path.join(SAGE_LOCAL, "lib")]
-    m.include_dirs = m.include_dirs + include_dirs
+    m.library_dirs = m.library_dirs + libgap_library_dirs + [os.path.join(SAGE_LOCAL, "lib")]
+    m.include_dirs = m.include_dirs + libgap_include_dirs + include_dirs
 
 
 #############################################
